File: LLD/LLD-2/src/TicTakToeGame/src/main/java/org/example/strategies/BotPlayingStrategy/EasyBotPlayingStrategy.java
package org.example.strategies.BotPlayingStrategy;

import org.example.models.Board;
import org.example.models.Cell;
import org.example.models.CellState;
import org.example.models.Move;

import java.util.List;

public class EasyBotPlayingStrategy implements BotPlayingStrategy{
    @Override
    public Move makeMove(Board board) {
        //In easy bot playing strategy we will make move in first available cell

        for(List<Cell> row : board.getBoard())
        {
            for(Cell cell : row)
            {
                if(cell.getCellState().equals(CellState.EMPTY))
                {
                    //Bot can make a move in this cell
                    return new Move(cell, null);
                }
            }
        }
        return null;
    }
}

File: LLD3/TicTacToe/TicTacToe/Strategies/PlayingStrategies/EasyBotPlayingStrategy.cs
using System;
using System.Collections.Generic;
using System.Globalization;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TicTacToe.Models;

namespace TicTacToe.Strategies.PlayingStrategies
{
    public class EasyBotPlayingStrategy : IBotPlayingStrategy
    {
        public Move MakeMove(Board board)
        {
            //In easy bot playing strategy bot will move in a first empty cell.
            for (int i = 0;i<board.Dimension;i++)
            {
                for(int j=0;j<board.Dimension;j++)
                {
                    Cell cell = board.Boards[i,j];
                    if(cell.State.Equals(CellState.Empty))
                    {
                        return new Move(null, cell);
                    }
                }
            }
            return null;
        }
    }
}
